app/columns/columns-modal.tsx

```diff
@@ -0,0 +1,36 @@
+import React from "react"
+import ReactDOM from "react-dom"
+import {useDispatch, useSelector} from "react-redux"
+import {CSSTransition} from "react-transition-group"
+import {XColumnChooserMenu} from "src/js/components/ColumnChooserMenu"
+import lib from "src/js/lib"
+import Modal from "src/js/state/Modal"
+
+export default function ColumnsModal() {
+  const dispatch = useDispatch()
+  const modal = useSelector(Modal.getName)
+  if (modal !== "columns") return null
+
+  const onClose = () => dispatch(Modal.hide())
+  const style = {
+    right: 0,
+    top: 0,
+    height: lib.win.getHeight()
+  }
+  return ReactDOM.createPortal(
+    <CSSTransition
+      in={true}
+      classNames="dim-portal-overlay"
+      timeout={{enter: 200}}
+      onClick={onClose}
+      appear
+    >
+      <div className="portal-overlay">
+        <div className="portal-item" style={style}>
```

Comment:
Is there any way we can style this all with just styled-components?

Reply:
I don't want this Portal Overlay to live much longer. This is the only place it's used, so I've added a deprecation comment. Next time we need to add features to the column picker, I'd like to delete this "portal" component. And at that time with Styled Components. Just didn't seem worth it to convert it.